cmd/util/ledger/migrations/atree_register_migration.go

```diff
@@ -0,0 +1,443 @@
+			if _, isADomainKey := domainsLookupMap[id.Key]; isADomainKey {
+				// this is expected. Move it to the new payloads
+				newPayloads = append(newPayloads, value)
+				continue
+			}
+
+			if _, ok := storageMapIds[id.Key]; ok {
+				// This is needed because storage map can be empty.
+				// Empty storage map only exists in old payloads because there isn't any element to migrate.
+				newPayloads = append(newPayloads, value)
+				continue
+			}
+
+			m.rw.Write(migrationProblem{
+				Address: mr.Address.Hex(),
+				Key:     id.String(),
+				Size:    len(mr.Snapshot.Payloads),
+				Kind:    "not_migrated",
+				Msg:     fmt.Sprintf("%x", value.Value()),
+			})
+
+			size, err := payloadSize(key, value)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get payload size: %w", err)
+			}
+
+			removedSize += size
+
+			// this is ok
+			// return nil, skippableAccountError
+		}
+	}
+
+	if statePayload == nil {
+		return nil, fmt.Errorf("state payload was not found")
+	}
+
+	// since some registers were removed, we need to update the storage used
+	if removedSize > 0 {
+		status, err := environment.AccountStatusFromBytes(statePayload.Value())
+		if err != nil {
+			return nil, fmt.Errorf("could not parse account status: %w", err)
+		}
+
+		status.SetStorageUsed(status.StorageUsed() - removedSize)
+
+		newPayload, err := newPayloadWithValue(statePayload, status.ToBytes())
+		if err != nil {
+			return nil, fmt.Errorf("cannot create new payload with value: %w", err)
+		}
+
+		statePayload = newPayload
+	}
+
+	newPayloads = append(newPayloads, statePayload)
+
+	return newPayloads, nil
+}
+
+func (m *AtreeRegisterMigrator) cloneValue(
+	mr *migratorRuntime,
+	value interpreter.Value,
+) (interpreter.Value, error) {
+
+	err := capturePanic(func() {
+		// force the value to be read entirely
+		value = value.Clone(mr.Interpreter)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return value, nil
+}
+
+// capturePanic captures panics and converts them to errors
+// this is needed for some cadence functions that panic on error
+func capturePanic(f func()) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			var stack [100000]byte
+			n := runtime2.Stack(stack[:], false)
+			fmt.Printf("%s", stack[:n])
+
+			switch x := r.(type) {
+			case runtime.Error:
+				err = fmt.Errorf("runtime error @%s: %w", x.Location, x)
+			case error:
+				err = x
+			default:
+				err = fmt.Errorf("panic: %v", r)
+			}
+		}
+	}()
+	f()
+
+	return
+}
+
+// convert all domains
+var domains = []string{
+	common.PathDomainStorage.Identifier(),
+	common.PathDomainPrivate.Identifier(),
+	common.PathDomainPublic.Identifier(),
+	runtime.StorageDomainContract,
+	stdlib.InboxStorageDomain,
+}
+
+var domainsLookupMap = map[string]struct{}{
+	common.PathDomainStorage.Identifier(): {},
+	common.PathDomainPrivate.Identifier(): {},
+	common.PathDomainPublic.Identifier():  {},
+	runtime.StorageDomainContract:         {},
+	stdlib.InboxStorageDomain:             {},
+}
```

Comment:
Given that Testnet already had Cap Cons enabled, those also have to be migrated, i.e. `stdlib.CapabilityControllerStorageDomain` needs to be added

```suggestion
	stdlib.CapabilityControllerStorageDomain,
}

var domainsLookupMap = map[string]struct{}{
	common.PathDomainStorage.Identifier():    {},
	common.PathDomainPrivate.Identifier():    {},
	common.PathDomainPublic.Identifier():     {},
	runtime.StorageDomainContract:            {},
	stdlib.InboxStorageDomain:                {},
	stdlib.CapabilityControllerStorageDomain: {},
}
```